### ISSUE TITLE
AddressFixups does not support AsmResolver.Symbol

### ISSUE THREAD

**Issue by BadRyuner** [Washi1337/AsmResolver#444]
### AsmResolver Version

5.3.0

### .NET Version

Any

### Operating System

Windows

### Describe the Bug

AddressFixups does not support AsmResolver.Symbol

### How To Reproduce

```csharp
ModuleDefinition module = ...;
NativeMethodBody native = ...;
DataSegment segment = new DataSegment(bytes);
native.AddressFixups.Add(new AddressFixup(somepos), AddressFixupType.Absolute64BitAddress, new Symbol(segment.ToReference())));

var image = module.ToPEImage(); // exception!
```

### Expected Behavior

Ability to use DataSegment as a symbol in NativeMethodBody

### Actual Behavior

Exception:
`System.AggregateException: "Construction of the PE image failed with one or more errors. (Symbols of type AsmResolver.Symbol are not supported.)"`

### Additional Context
https://github.com/Washi1337/AsmResolver/blob/master/src/AsmResolver.DotNet/Code/Native/NativeSymbolsProvider.cs#L26-L32

_No response_

**Comment by Washi1337** [comment]
A potential fix for this is in #444. Could I ask you once again to verify this works for your use-case?

Your bug reports are highly appreciated btw!

**Comment by BadRyuner** [comment]
I had already moved far away from this idea, so for the test I used [this](https://github.com/BadRyuner/UnmanagedString) project with a slight modification. There are no errors, everything works perfectly.
